contracts/service-registry/src/contract.rs

```diff
@@ -272,6 +276,145 @@ mod test {
         ));
     }
 
+    #[test]
```

Comment:
can you split this test into multiple or find another way to compact it? It's so long that it's hard to understand what's going on